core/src/main/java/io/aiven/klaw/helpers/db/rdbms/InsertDataJdbc.java

```diff
@@ -380,13 +381,21 @@ public String insertIntoRegisterUsers(RegisterUserInfo userInfo) {
     Optional<UserInfo> userNameExists = userInfoRepo.findById(userInfo.getUsername());
     if (userNameExists.isPresent()) return "Failure. User already exists";
 
-    // STAGING status comes from AD users
-    RegisterUserInfo userRegistration =
-        registerInfoRepo.findFirstByUsernameAndStatusIn(
-            userInfo.getUsername(),
-            List.of(NewUserStatus.PENDING.value, NewUserStatus.STAGING.value));
-    if (userRegistration != null) {
-      return "Failure. Registration already exists";
+    Optional<RegisterUserInfo> optionalUserRegistration =
+        registerInfoRepo.findByUsername(userInfo.getUsername());
+    if (optionalUserRegistration.isPresent()) {
```

Comment:
If the User Registration is not present should we return "Failure. Registration doesn't exist"

Reply:
We should create a new one if not present which is done at line 401.